File: CartrawlerSDK/UI/CartrawlerSDK/BookingSummaryViewController.m
//
//  BookingSummaryViewController.m
//  CartrawlerSDK
//
//  Created by Lee Maguire on 12/07/2016.
//  Copyright © 2016 Cartrawler. All rights reserved.
//

#import "BookingSummaryViewController.h"
#import <CartrawlerAPI/CTVendor.h>
#import "CartrawlerSDK+NSDateUtils.h"
#import "CTButton.h"
#import "CTLabel.h"
#import "CTAppearance.h"
#import "CTPaymentSummaryDataSource.h"

@interface BookingSummaryViewController ()

@property (nonatomic, weak) IBOutlet UITableView *tableView;
@property (nonatomic, strong) CTPaymentSummaryDataSource *dataSource;

@end

@implementation BookingSummaryViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    if (!self.dataSource) {
        _dataSource = [[CTPaymentSummaryDataSource alloc] init];
    }
    
    self.tableView.dataSource = self.dataSource;
    self.tableView.delegate = self.dataSource;
    self.tableView.rowHeight = UITableViewAutomaticDimension;
    self.tableView.estimatedRowHeight = 100;
    
}

- (IBAction)back:(id)sender
{
    [self.navigationController popViewControllerAnimated:YES];
}

- (void)viewWillAppear:(BOOL)animated
{
    [super viewWillAppear:animated];
    
    [self.tableView reloadData];
}

- (void)setData:(CarRentalSearch *)search
{
    if (self.dataSource) {
        [self.dataSource setData:search];
    } else {
        _dataSource = [[CTPaymentSummaryDataSource alloc] init];
        [self.dataSource setData:search];
    }
}

@end

File: CartrawlerSDK/CartrawlerTestSwift/CartrawlerTestSwift/BookingSummaryViewController.swift
//
//  BookingSummaryViewController.swift
//  CartrawlerTestSwift
//
//  Created by Lee Maguire on 16/09/2016.
//  Copyright © 2016 Cartrawler. All rights reserved.
//

import UIKit

class BookingSummaryViewController: CTViewController {

    @IBOutlet weak var vehicleImageView: UIImageView!
    @IBOutlet weak var vehicleNameLabel: UILabel!
    @IBOutlet weak var supplierNameLabel: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        vehicleNameLabel.text = search.selectedVehicle.vehicle.makeModelName
        supplierNameLabel.text = search.selectedVehicle.vendor.name
        
        //called when pushToDestination() fails so you can display an error to user
        dataValidationCompletion = { success, errorMessage in
            
        }
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func back(_ sender: AnyObject) {
        _ = navigationController?.popViewController(animated: true)
    }
    
    @IBAction func next(_ sender: AnyObject) {
        pushToDestination()
    }
    
}
